src/store/messagesStore.js

```diff
@@ -628,6 +629,37 @@ const actions = {
 		}
 	},
 
+	/**
+	 * Edit a message text
+	 *
+	 * @param {object} context default store context;
+	 * @param {object} payload payload;
+	 * @param {string} payload.token The conversation token
+	 * @param {string} payload.messageId The message id
+	 * @param {string} payload.updatedMessage The modified text of the message / file share caption
+	 */
+	async editMessage(context, { token, messageId, updatedMessage }) {
+		const message = Object.assign({}, context.getters.message(token, messageId))
+		context.commit('addMessage', {
+			token,
+			message: { ...message, message: updatedMessage },
+		})
```

Comment:
Why don't you add it after EditMessage request? so you don't need to restore upon failure and it is only called in successful.